### DISCUSSION TITLE
Recommendations Needed: Should I create one Big MindMap to contain all my data or multiple ones on Topics?

### DISCUSSION THREAD

**Discussion by cricrazy** [freeplane/freeplane#1063] [Help]
First of all, Freeplane is my daily driver. This is an amazing software. I have been using it for a while now and have really been blessed by it. I do have a question for the community, more like recommendations. 

As I create more and more information through the mindmaps, I am getting to a point where it seems an information overload. I started with few big mindmaps, but as it grew, I started to create separate ones for specific topic (area of my life). But now, I am having hard time finding the information in multiple of them.

### **Do you guys recommend, having one gigantic mindmap that includes all and then use Search to find to the sub topics and add or edit them?**

Is there a limit to size of an individual mindmap, where it gets too big to handle? I usually do not have lots of images, but I occasionally add image, if needed in my nodes. 

### **Can you please Share Your experience, lessons learned, ideas?** 

I understand that this can be just a personal preference, but I have always learnt a lot from these community suggestions/recommendations.  So , really appreciate this community. I have used SourceForge before and have received very active community responses. 

Thank you.

**Comment by Illegit** [comment]
I still have this kind of doubt, but...
In my point of view that choice depends on the RAM of your pc 
At home I have a 8GB ram (1GB dedicated to FP), I use a lot of smaller maps, with some bigger e.g.:

- one for personal administration
- one for calendar and activities schedule (I do not use GTD, much better some maps with tools provided to me by this forum users (again many thanks to all of them))
- one for FP instructions and solutions, 
- 4 different ones for music (I have only 8GB) therefore I have chosen where to use jumper (working only in single map) and where to use other searching tools (working on all of the opened maps)

At work: I have much bigger maps with plenty of formulas, clones and conditional styles (and a lot of images and links) and I try to group things in some maps only in order to use jumper as much as possible. But for this kind of usage I have dedicated 6GB RAM only to FP.

Hope it helps you to find an affordable solution...

**Reply by amitchristianYT** [reply]
Cool. Thanks for the feedback. I am curious what do you have in music mindmap? Also, can you please share if possible, how you use Calendar/activities schedule in Mindmap? Thanks.

**Comment by sifran-github** [comment]
I support a balanced approach: not everything in one map, but also not divided into too many maps.

If you have one specific topic that has a lot of information (maybe for example because its your main work), put it into 1 or maybe a few mind maps. keep them together (folder?) and then you know where to find information when you need it.
If you have many topics that have with a bit of information in them, for example random topics that you search online, just batch them together into one map.

**Reply by cricrazy** [reply]
Great suggestion. Thanks.

**Comment by ldmpub** [comment]
Hi,

Very interesting topic 👍.

Usually I'm trying to use several little maps. Yet I have started my first big map for professional reasons ... but I must confess that it's hard to use it (even with the jumper Add-on). Here are the stats for this "too big map":

![image](https://user-images.githubusercontent.com/1452146/228066401-0d0b36db-5132-4e74-8107-13e486c8b2f3.png)

Sorry I'm running Freeplane in french ... translation: Total then Nb of last level nodes then Nb of nodes starting from the root node ...

Note: I'm quite sure this map will be considered as a little one by many advanced Freeplane users 😉but I have other items to add and I'm wondering if I will split them elsewhere...

**Reply by cricrazy** [reply]
Thanks. How did you get the stats for the map? I will run mine and see what it shows. Thanks.

**Reply by EdoFro** [reply]
> Note: I'm quite sure this map will be considered as a little one by many advanced Freeplane users 😉but I have other items to add and I'm wondering if I will split them elsewhere...

I just looked at the map I am working on (it's from my work at one client):
![image](https://user-images.githubusercontent.com/35700575/228315938-47212398-3943-4f12-bbd9-297812d0de30.png)

and this is one of my big maps, but by far not the biggest one.

I have no problem navigating this. I have to be consistent on how I organize things. It don't need to be perfect organized, just good enough, but you have to be consistent how you classify things in it. 
Also, you must think in advance how you will search for something. This way when you add something that you think you will need in the future, you can add special words, styles or icons to easily find them.

In my life I have learned to place things not where there logically belong, but where I think I will search for them later. And it works better.  Sometimes both are the same place, but sometimes not and I waste a lot of time searching and only after I find it I realized how logical and obvious its location was. But next time I waste as much time as the one before! 
That's why I now, if I waste time searching for something in my house, when I find it, then I put it in the first place I looked for it. 

jm2c
edo

**Reply by bepolymathe** [reply]
Hi all, 

> I have no problem navigating this. 

![image](https://user-images.githubusercontent.com/29939805/228337338-311367a5-dc6e-4923-9bb6-9c64a7cdd843.png)

Same for me...

**Reply by cricrazy** [reply]
@EdoFro I absolutely agree with you. I have been also learning how to organize the information that I myself, will eventually need sometime in future and how would I search for that information. Really appreciate your feedback. This is why I love Freeplane, because it almost gives an insight into one's thought process, which is super useful for others to learn from or replicate. 

I really think, that Freeplane community is the powerhouse of the ideas and implementation of the data information. I wish we can share our knowledge (of what works and what doesn't) to maximize the Freeplane's potential. 

Thanks again.

**Comment by Lotus49b** [comment]
I find that maps become too big, especially on a laptop display.

I break the maps up into smaller project or idea-specific maps, and create a "Map of maps" with hyperlinks to all maps.

It's a bit of work to set up initially, but it makes it easy to find exactly what you need quickly. Also, the workflow to add new maps to the MoM is quick and easy when you get into the habit.

- Make MoM
- Create node with target map description
- Ctrl-Shift-K to link to target
- Drag nodes around until you have created hierarchy and sections that make sense (this is where the new node layouts are great)

Hope this helps.

**Reply by cricrazy** [reply]
Thanks. Appreciate your feedback. I have tried to put the maps into manageable pieces, but it has not worked out for me, as I look for the info later. Thanks.

**Comment by EdoFro** [comment]
I have a similar approach as @sifran-github,

> I support a balanced approach: not everything in one map, but also not divided into too many maps.

here some "principles" I try to follow:
- I assign Freeplane enough memory to work properly ( I don't want to limit my maps because of computer lacking of memory)
- I only create a 'new' map when it's an indepent topic that deserves its own map
- I allways (try to) create new maps by using the command `File->Move branch to new map… (Alt+Shift+A)`. 
This way:
  - there is some relation between maps (you can navigate up and down)
  - I export it as new map only when it has some structure and information in it (I try to avoid little mindmaps)
  - they inherit all styles and conditional style rules automatically. I can work with them as usual.
- I have a central map that works as a "root map". All other maps were originated from this one or one of its descendant maps.
Teorically I can create a "map crawler" using groovy script. I haven't yet, but it's in my projects mindmap.

Hope it helps,
edo

**Reply by arresse** [reply]
I basically use one large map but use View/New Map View to work in different areas rather than jumping around. However I do use a system of internal links to get to pretty much anywhere in the map with 1-3 clicks maximum.

I'm interested i what edo said about allocating memory to FP. How exactly is this done? I have a Windows 8.1 PC with 8GB RAM.

Also are their any tips n tricks to making the most of your RAM with FP?

**Reply by EdoFro** [reply]
in windows, there are two .ini files in the base directory of Freeplane:
- freeplane.l4j.ini
- freeplaneConsole.l4j.ini

You have to edit them (with notepad or similar) and change the `-Xmx` line.

Look at the information [in the old wiki](https://docs.freeplane.org/attic/old-mediawiki-content/Troubleshooting.html#freeplane-throws-out-of-memory-exception)

**Reply by Illegit** [reply]
And I do the same even for the file freeplane.bat.: eg.: delete 512b and write 1g.

**Reply by EdoFro** [reply]
I use “2G”, but I don’t really know if it’s the optimum. It works.

**Reply by arresse** [reply]
Thank you for your replies. Are there also any general tips for making the most of resources with FP?

**Reply by EdoFro** [reply]
I think this question deserves its own discussion, don’t you?

**Reply by Bricklam** [reply]
>     * I have a central map that works as a "root map". All other maps were originated from this one or one of its descendant maps.
>       Teorically I can create a "map crawler" using groovy script. I haven't yet, but it's in my projects mindmap.

I like the idea of "map crawler" script it sounds interesting, EdoFro. I hope to be able to share it with the community if you had the opportunity to accomplish it.
Thanks

**Reply by ldmpub** [reply]
> I think this question deserves its own discussion, don’t you?

I agree, please open a separate discussion for this technical topic starting with how to optimize computer RAM used by Freeplane ...

**Comment by grotesque** [comment]
how can one search on multiple maps at once?

**Comment by cps-viasat** [comment]
I find that I like to have a central To-Do map that gets me started on what's relevant, and is quickly and easily updated, but then I link off Projects™️ and larger task groupings if they're going to run for a while. That way I can un-link them when they're no longer at the top of the pile, but the separate MM is still there if I have to go back to look at notes.
